hooks/pre_validations/31-puppet-agent-oauth.rb

```diff
@@ -1,7 +1,7 @@
 unless app_value(:noop)
   if File.exist?('/opt/puppetlabs/puppet/bin/ruby') && execute_command("rpm -q puppet-agent-oauth", false, false)
-    unless execute_command("/opt/puppetlabs/puppet/bin/ruby -e \"require 'oauth'\"", false, false)
-      execute("yum -y reinstall puppet-agent-oauth", false, true)
+    unless execute_command_nonscl("/opt/puppetlabs/puppet/bin/ruby -e \"require 'oauth'\"", false, false)
+      execute_command_nonscl("yum -y reinstall puppet-agent-oauth", false, true)
```

Comment:
Is the SCL context important here? If so, it indicates https://github.com/theforeman/foreman-packaging/blob/024d0a3d7a2b2a3cad98f9e89a12cb458d30593f/packages/foreman/puppet-agent-oauth/puppet-agent-oauth.spec#L32-L50 isn't properly isolated.

Reply:
Yes. If I execute it in SCL context, it will happily load the SCL oauth gem.

Reply:
"nice"

Reply:
I wonder if the installer should _always_ execute external commands w/o SCL context.

Reply:
Looking at what's executed, yes, I think that makes sense.

Reply:
@ehelms what do you think?

Also, shall we put that in this PR or later?

Reply:
If we decide to non-scl everything I think it makes sense to do it here. Introducing a method and removing it later feels bad.